service/storage_service.cc

```diff
@@ -3969,6 +3969,12 @@ future<> storage_service::on_alive(gms::inet_address endpoint, gms::endpoint_sta
     const auto& tm = get_token_metadata();
     const auto tm_host_id_opt = tm.get_host_id_if_known(endpoint);
     slogger.debug("endpoint={}/{} on_alive: permit_id={}", endpoint, tm_host_id_opt, pid);
+
+    if (_raft_topology_change_enabled) {
+        slogger.debug("ignore on_alive since topology changes are using raft");
+        co_return;
+    }
+
     bool is_normal_token_owner = tm_host_id_opt && tm.is_normal_token_owner(*tm_host_id_opt);
     if (is_normal_token_owner) {
         co_await notify_up(endpoint);
```

Comment:
Do we need `notify_up` in raft topology mode? I see `cql_server::event_notifier::on_up` is doing something meaningful.

Reply:
We probably need. We don't ignore `storage_service::on_dead`, which initiates calls to `on_down`. I suppose that calling `on_down` first, and then not calling `on_up` when the node comes back to life would cause something bad on the client side.

What confuses me is that we also have `notify_left` and it looks like it is never called in the Raft-based topology. And `cql_server::event_notifier::on_leave_cluster` also does something meaningful.

Reply:
there are three implementations of `service::endpoint_lifecycle_subscriber`:
```
service_level_controller
    abort some processes through on_leave_cluster

storage_proxy
    purges some data structures through on_leave_cluster and on_down

cql_server::event_notifier
    send notifications to drivers
```
all of them are not dead code and should be notified properly.

`notify_left` is called from `storage_service::excise`
```
notify_left <- excise <- handle_state_left
  called from storage_service::on_change

  there is already if (_raft_topology_change_enabled) return;

notify_left <- excise <- handle_state_removed
   called from storage_service::on_change

   there is no if (_raft_topology_change_enabled) return;

notify_left <- excise <- storage_service::removenode

notify_left <- excise <- storage_service::force_remove_completion
```

* should we call `notify_left` from `raft_removenode`?

*  `if (_raft_topology_change_enabled) return;` was added to `storage_service::handle_state_left` here
https://github.com/scylladb/scylladb/commit/f3bd7e9b8c610851dad07838c7bc43e99e1085c7, should we restore `notify_left` call in it?

@gleb-cloudius

Reply:
We discussed this on a daily:
  * there is a separate task in our raft-topology todo list - check that these `notify` methods are called correctly, fix if not;
  * this is orthogonal to the current PR. I suggest preserve the current behaviour for now - do not touch `token_metadata` if `_raft_topology_change_enabled`, but keep the `notify` calls.

Reply:
> I suggest preserve the current behaviour for now - do not touch `token_metadata` if `_raft_topology_change_enabled`, but keep the `notify` calls.

I did exactly that in `on_alive`.